specification/security/resource-manager/Microsoft.Security/preview/2023-02-01-preview/securityConnectors.json

```diff
@@ -0,0 +1,1494 @@
+        },
+        "policyAgentAutoProvisioningFlag": {
+          "type": "boolean",
+          "description": "Is Policy Kubernetes agent auto provisioning enabled"
+        }
+      }
+    },
+    "cspmMonitorGithubOffering": {
+      "type": "object",
+      "description": "The CSPM monitoring for github offering",
+      "x-ms-discriminator-value": "CspmMonitorGithub",
+      "allOf": [
+        {
+          "$ref": "#/definitions/cloudOffering"
+        }
+      ],
+      "properties": {}
+    },
+    "cspmMonitorAzureDevOpsOffering": {
+      "type": "object",
+      "description": "The CSPM monitoring for AzureDevOps offering",
+      "x-ms-discriminator-value": "CspmMonitorAzureDevOps",
+      "allOf": [
+        {
+          "$ref": "#/definitions/cloudOffering"
+        }
+      ],
+      "properties": {}
+    },
+    "defenderCspmAwsOffering": {
+      "type": "object",
+      "description": "The CSPM P1 for Aws offering",
+      "x-ms-discriminator-value": "DefenderCspmAws",
+      "allOf": [
+        {
+          "$ref": "#/definitions/cloudOffering"
+        }
+      ],
+      "properties": {
+        "vmScanners": {
+          "type": "object",
+          "description": "The Microsoft Defender for Server VM scanning configuration",
+          "properties": {
+            "enabled": {
+              "type": "boolean",
+              "description": "Is Microsoft Defender for Server VM scanning enabled"
+            },
+            "configuration": {
+              "type": "object",
+              "description": "configuration for Microsoft Defender for Server VM scanning",
+              "properties": {
+                "cloudRoleArn": {
+                  "type": "string",
+                  "description": "The cloud role ARN in AWS for this feature"
+                },
+                "scanningMode": {
+                  "type": "string",
+                  "description": "The scanning mode for the vm scan.",
+                  "enum": [
+                    "Default"
+                  ],
+                  "x-ms-enum": {
+                    "name": "scanningMode",
+                    "modelAsString": true
+                  }
+                },
+                "exclusionTags": {
+                  "type": "object",
+                  "additionalProperties": {
+                    "type": "string"
+                  },
+                  "description": "VM tags that indicates that VM should not be scanned"
+                }
+              }
+            }
+          }
+        },
+        "dataSensitivityDiscovery": {
+          "type": "object",
+          "description": "The Microsoft Defender Data Sensitivity discovery configuration",
+          "properties": {
+            "enabled": {
+              "type": "boolean",
+              "description": "Is Microsoft Defender Data Sensitivity discovery is enabled"
+            },
```

Comment:
ARM recommends using enums over booleans for new properties, as it can help to ensure future proof APIs.

Standard guidance is: replace boolean/switch properties with better enum whenever possible.
A boolean will forever have two valid values (true or false). A string enum is more flexible and future proof option because it allows additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].. Also, properties should always ideally provide more meaningful values just than True and False, which are meaningless without context. For example two boolean switches "isTypeA" and "isTypeB" can be replaced with one enum "type": [A, B, DefaultType]. 
Note: do NOT just define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its still confusing, thanks to the boolean value names. Hint: Try to think of enum value names that look meaningful even without context like 'Unencrypted', 'Encrypted', 'TlsEncrypted' which you can use instead.

[Conversely: do not do breaking changes by change boolean properties in use by customers with existing API versions.]

Reply:
I understand - and this is a good approach. 
but in this case, specifically, the property is "enabled" - i don't see any future use for more additional values. 
anyhow - will take care of this starting next API version since we are short in time (hope this is ok) 

thanks a lot

Reply:
Yes, its up to you.